<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
    <title>Training</title>

    <!-- Bootstrap -->
    <link rel="stylesheet" href="/css/bootstrap.min.css">
    <link rel="stylesheet" href="/css/font-awesome.min.css">
    <link href="/css/tops.css" rel="stylesheet">
    <link href='http://fonts.googleapis.com/css?family=Shadows+Into+Light' rel='stylesheet' type='text/css'>

    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
  </head>
  <body>
    <nav class="navbar navbar-inverse tops-navbar">
      <div class="first_page container">
        <!-- Brand and toggle get grouped for better mobile display -->
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="/index.html">MYOP</a>
        </div>

        <!-- Collect the nav links, forms, and other content for toggling -->
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
          <ul class="nav navbar-nav">
            <li><a href="/download.html">Download</a></li>
            <li><a href="/documentation.html">Documentation</a></li>
          </ul>

          <ul class="nav navbar-nav navbar-right">
            <li><a href="https://github.com/myopdev/myop"><span class="glyphicon fa fa-github-alt" aria-hidden="true"></span></a></li>
            <li><a href="/index.html#contact"><span class="glyphicon fa fa-envelope-o" aria-hidden="true"></span></a></li>
          </ul>

        </div><!-- /.navbar-collapse -->
      </div><!-- /.first_page container-fluid -->
    </nav>

    <div class="wrapper">
    <div class="container tops-doc-body">
  <div class="row">
    <div class="col-md-3 tops-doc">
      <h1>User Guide</h1>
      <ol>
        
        <li><a href="/guides/01-setup.html">Setup</a></li>
        
        <li><a href="/guides/02-prediction.html">Prediction</a></li>
        
        <li><a href="/guides/03-training.html">Training</a></li>
        
        <li><a href="/guides/04-tutorial.html">Tutorial</a></li>
        
        <li><a href="/guides/05-pre-trained-models.html">Pre-Trained Models</a></li>
        
        <li><a href="/guides/06-accuracy.html">Accuracy</a></li>
        
      </ol>
    </div>
    <div class="col-md-9 tops-page">

      <h1 id="training">Training</h1>

<p>You can train your own models, that is, you can use your sequences to train a new predictor. For this you need a (multi) fasta file containing the genomic sequence(s) and a description of the genes in a <strong>GTF</strong> file (for more details on GTF format, use this <a href="http://www.ensembl.org/info/website/upload/gff.html">link</a>), which is a standard format for training gene predictors.</p>

<p><strong>GTF</strong> files should describe all the CDSâ€™s of each gene and the positon of the start and stop codons. You can check your gtf file using the <a href="http://mblab.wustl.edu/software.html">validate_gtf.pl</a> script.</p>

<p>If you have your genomic sequences in the file named <code class="highlighter-rouge">genome.fasta</code> and your <strong>GTF</strong> entries in the file named  <code class="highlighter-rouge">genome.gtf</code>, you can train your predictor using the command:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>myop-train -g genome.gtf -f genome.fa -o my_org
</code></pre>
</div>

<p>in this case, the training procedure will create a directory named <code class="highlighter-rouge">my_org</code> with the complete description of the model. All names (fasta file, gtf file, model directory)  can be changed by the user.</p>

<p>After training you can use the model as is (see the instructions for running gene predictions),</p>

<div class="highlighter-rouge"><pre class="highlight"><code>myop-predict -l /Path/To/my_org/ -f seq.fa
</code></pre>
</div>

<p>or you can add the new model to your installation. To do this you should use the command:</p>

<div class="highlighter-rouge"><pre class="highlight"><code>myop-add-genome /Path/To/my_org/
</code></pre>
</div>

<p>With this you can run predicitons with <strong>MYOP</strong> by just using the name you created for your model.</p>

<div class="highlighter-rouge"><pre class="highlight"><code>myop-predict -g my_org -f seq.fa
</code></pre>
</div>


      <p class="text-center">
        
          <a href="/guides/02-prediction.html">prev</a>
        
        |
        
          <a href="/guides/04-tutorial.html">next</a>
        
      </p>
    </div>
  </div>
</div>

    </div>

    <div class="tops-footer">
      <div class="first_page container">
        <p class="text-center">Copyright 2016 MYOP</p>
      </div>
    </div>

    <!-- Bootstrap core JavaScript
    ================================================== -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
    <script src="/js/bootstrap.min.js"></script>
  </body>
</html>